packages/core-contracts/contracts/utils/ERC20.sol

```diff
@@ -0,0 +1,93 @@
+//SPDX-License-Identifier: UNLICENSED
+
+pragma solidity ^0.8.0;
+
+import "../interfaces/IERC20.sol";
+
+abstract contract ERC20 is IERC20 {
+    string public override name;
+
+    string public override symbol;
+
+    uint8 public immutable override decimals;
+
+    mapping(address => uint256) public override balanceOf;
+
+    mapping(address => mapping(address => uint256)) public override allowance;
```

Comment:
Should we stick to the pattern of always using private variables and exposing them via our own getters?
I.e. rename this to `private _allowances`

Reply:
So this si something I thought a bit about. I went for code efficiency. If we decide to go with this approach we should follow it in all contracts and we will end up bloating the contract with getters.